x/wasm/keeper/msg_server_integration_test.go

```diff
@@ -333,3 +333,106 @@ func TestMigrateContract(t *testing.T) {
 		})
 	}
 }
+
+func TestExecuteContract(t *testing.T) {
+	wasmApp := app.Setup(false)
+	ctx := wasmApp.BaseApp.NewContext(false, tmproto.Header{Time: time.Now()})
+
+	var (
+		myAddress       sdk.AccAddress = make([]byte, types.ContractAddrLen)
+		_, _, otherAddr                = testdata.KeyTestPubAddr()
+	)
+
+	specs := map[string]struct {
+		addr   string
+		expErr bool
+	}{
+		"adress can execute a contract": {
+			addr:   myAddress.String(),
+			expErr: false,
+		},
+		"other address cannot execute a contract": {
+			addr:   otherAddr.String(),
+			expErr: true,
+		},
+	}
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			xCtx, _ := ctx.CacheContext()
+			// setup
+			_, _, sender := testdata.KeyTestPubAddr()
+			msg := types.MsgStoreCodeFixture(func(m *types.MsgStoreCode) {
+				m.WASMByteCode = hackatomContract
+				m.Sender = sender.String()
+			})
```

Comment:
If you want to customize the msg, I think it is better to use `types.MsgStoreCode
````suggestion
			storeMsg := &types.MsgStoreCode{
				Sender:            sender.String(),
				WASMByteCode:          hackatomContract,
				InstantiatePermission: &types.AllowEverybody,
			}
```

Reply:
The same thing can be done with `types.MsgStoreCodeFixture`, so there is no need to change it.